Automattic/jetpack
# Social: Refactor feature option storing

This change refactors how we store our options for SIG and auto conversion. We moved to use the register_setting function, this way we need no Rest controllers.
It's easier to review this following the commits.

FEATURE BRANCH OTHER PRs:
* Social: Remove deprecated feature files #34114
* Social: Refresh store on Publicize module toggle via 1 call #34142


## Proposed changes:

* f37dd1a70f2259ef236166a183b2eec1c1c1f2c5 - Added the new settings file and tests for it.
* d00d07e9d311f341e33b250b224c18a1f73cb359 - Updated the initial state, so it uses the new object. Also changed every usage where SIG needed the default template
* 012eb2ce2482f79ea11ff7a40c6ba08d38696a39 - This commit updates the UI parts to match the new data structure, and also updates the store so it calls the core wp/v2 endpoint to retrieve and update the settings.

### Other information:


## Jetpack product discussion


https://github.com/orgs/Automattic/projects/733/views/2?pane=issue&itemId=44049666
## Does this pull request change what data or activity we track or use?


No.
## Testing instructions:






This probably needs a lot of testing to see that everything is working. You can run the tests on the backend with `composer run test-php` from the `publicize` folder.

To test the flow you can do the following:
* Have both Jetpack and Jetpack Social enabled. Open up The Jetpack Settings, and go to the sharing section. On another tab open up the Social admin page.
  * Toggle on/off SIG and the conversion in one of the pages, then refreshing the other page should reflect the changes. Try to change the template too.
  * Make sure that turning the main module off, refreshing, then turning it on shows the correct values.
* Have the conversion on, then create a new post. The default state of SIG in the editor should match the state of the toggle in the admin page. 

* Clear the dismissed notices with `jetpack docker wp option delete jetpack_social_dismissed_notices`
* Open up a new post with the conversion on, then upload an invalid image that can be converted. You should see the notice that the image will be converted. 
  * Click on change settings, then turn the conversion off
  * Switch back to the other tab, the page should reflect the changes without needing to refresh it
